src/recursive_sorting/recursive_sorting.py

```diff
@@ -1,17 +1,32 @@
 # TO-DO: complete the helpe function below to merge 2 sorted arrays
-def merge( arrA, arrB ):
-    elements = len( arrA ) + len( arrB )
-    merged_arr = [0] * elements
-    # TO-DO
-    
-    return merged_arr
+def merge(arrA, arrB):
+
+    left_index, right_index = 0, 0
+    result = []
+    while left_index < len(arrA) and right_index < len(arrB):
+        if arrA[left_index] < arrB[right_index]:
+            result.append(arrA[left_index])
+            left_index += 1
+        else:
+            result.append(arrB[right_index])
+            right_index += 1
+
+    result += arrA[left_index:]
+    result += arrB[right_index:]
+    return result
 
 
 # TO-DO: implement the Merge Sort function below USING RECURSION
-def merge_sort( arr ):
-    # TO-DO
+def merge_sort(arr):
+    if len(arr) <= 1:  # base case
+        return arr
 
-    return arr
+    # divide array in half and merge sort recursively
+    half = len(arr) // 2
```

Comment:
nice nice nice! this is one of the more difficult recursive sorting methods out there. great job solving it!